File: src/core/parsers/docx.ts
import * as fs from 'fs'
import { RawBRD } from '../models/brd'

export async function parseDocx(filePath: string): Promise<RawBRD> {
  // eslint-disable-next-line @typescript-eslint/no-require-imports
  const mammoth = require('mammoth')
  const buffer = fs.readFileSync(filePath)
  const result = await mammoth.extractRawText({ buffer })
  return {
    text: result.value,
    source: filePath,
    fileType: 'docx'
  }
}

File: pythonlibrary/aibrd/parsers/docx.py
from ..models.brd import RawBRD


def parse_docx(file_path: str) -> RawBRD:
    from docx import Document
    doc = Document(file_path)
    text = "\n\n".join(p.text for p in doc.paragraphs if p.text.strip())
    return RawBRD(text=text, source=file_path, file_type="docx")
